standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnStore.java

```diff
@@ -541,6 +554,15 @@ Set<CompactionInfo> findPotentialCompactions(int abortedThreshold, long abortedT
   @RetrySemantics.CannotRetry
   void markCleaned(CompactionInfo info) throws MetaException;
 
+  /**
+   * This will remove an aborted entries from TXN_COMPONENTS table after
+   * the aborted directories are removed from the filesystem.
+   * @param info info on the aborted directories cleanup that needs to be removed
+   * @throws MetaException
+   */
+  @RetrySemantics.CannotRetry
+  void markCleanedForAborts(AcidTxnInfo info) throws MetaException;
```

Comment:
I wouldn't create a separate API just for that

Reply:
Removed the API. Extended `markCleaned` with a boolean `isAbortOnly` flag. Done.